stl/inc/format

```diff
@@ -0,0 +1,2734 @@
+        _Callbacks._On_width(_Value);
+    } else if (*_Begin == '{') {
+        ++_Begin;
+        if (_Begin != _End) {
+            _Begin = _Parse_arg_id(_Begin, _End, _Width_adapter<_CharT, _Callbacks_type>{_Callbacks});
+        }
+        if (_Begin == _End || *_Begin != '}') {
+            throw format_error("Invalid format string.");
+        }
+        ++_Begin;
+    }
+    return _Begin;
+}
+
+template <class _CharT, _Parse_precision_callbacks<_CharT> _Callbacks_type>
+_NODISCARD constexpr const _CharT* _Parse_precision(
+    const _CharT* _Begin, const _CharT* _End, _Callbacks_type&& _Callbacks) {
+    ++_Begin;
+    _CharT _Ch = '\0';
+    if (_Begin != _End) {
+        _Ch = *_Begin;
+    }
+
+    if ('0' <= _Ch && _Ch <= '9') {
+        int _Precision = 0;
+        _Begin         = _Parse_nonnegative_integer(_Begin, _End, _Precision);
+        _Callbacks._On_precision(_Precision);
+    } else if (_Ch == '{') {
+        ++_Begin;
+        if (_Begin != _End) {
+            _Begin = _Parse_arg_id(_Begin, _End, _Precision_adapter<_CharT, _Callbacks_type>{_Callbacks});
+        }
+
+        if (_Begin == _End || *_Begin != '}') {
+            throw format_error("Invalid format string.");
+        }
+    } else {
+        throw format_error("Missing precision specifier.");
+    }
+    return _Begin;
+}
+
+template <class _CharT, _Parse_spec_callbacks<_CharT> _Callbacks_type>
+_NODISCARD constexpr const _CharT* _Parse_format_specs(
+    const _CharT* _Begin, const _CharT* _End, _Callbacks_type&& _Callbacks) {
+    if (_Begin == _End || *_Begin == '}') {
+        return _Begin;
+    }
+
+    _Begin = _Parse_align(_Begin, _End, _Callbacks);
+    if (_Begin == _End) {
+        return _Begin;
+    }
+
+    switch (*_Begin) {
+    case '+':
+        _Callbacks._On_sign(_Sign::_Plus);
+        ++_Begin;
+        break;
+    case '-':
+        _Callbacks._On_sign(_Sign::_Minus);
+        ++_Begin;
+        break;
+    case ' ':
+        _Callbacks._On_sign(_Sign::_Space);
+        ++_Begin;
+        break;
+    default:
+        break;
+    }
+
+    if (_Begin == _End) {
+        return _Begin;
+    }
+
+    if (*_Begin == '#') {
+        _Callbacks._On_hash();
+        if (++_Begin == _End) {
+            return _Begin;
+        }
+    }
+
+    if (*_Begin == '0') {
+        _Callbacks._On_zero();
+        if (++_Begin == _End) {
+            return _Begin;
+        }
+    }
+
+    _Begin = _Parse_width(_Begin, _End, _Callbacks);
+    if (_Begin == _End) {
+        return _Begin;
+    }
+
+    if (*_Begin == '.') {
+        _Begin = _Parse_precision(_Begin, _End, _Callbacks);
+        if (_Begin == _End) {
+            return _Begin;
+        }
+    }
+
+    if (*_Begin == 'L') {
+        _Callbacks._On_localized();
+        if (++_Begin == _End) {
+            return _Begin;
+        }
+    }
+
+    // If there's anything remaining we assume it's a type.
+    if (*_Begin != '}') {
+        _Callbacks._On_type(*_Begin++);

```

Comment:
Nitpick: Can we move the post decrement out of the function argument? It is easy to miss and different than all other places
```suggestion
        _Callbacks._On_type(*_Begin);
        ++_Begin;
```

Reply:
Agreed, good catch. I checked all of the other postincrements in `<format>` and this was the only atypical occurrence.